modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/OpenAPIResourceTest.java

```diff
@@ -267,6 +273,56 @@ public void testGetOpenAPIWithActions() throws Exception {
 			"expected_openapi_actions_object_action.json", _objectDefinition);
 	}
 
+	@Test
+	public void testGetOpenAPIWithSystemObjectRelationship() throws Exception {
+		_user = TestPropsValues.getUser();
+
+		_userSystemObjectDefinitionManager =
+			_systemObjectDefinitionManagerRegistry.
+				getSystemObjectDefinitionManager("User");
+
+		_userSystemObjectDefinition =
+			_objectDefinitionLocalService.fetchSystemObjectDefinition(
+				_userSystemObjectDefinitionManager.getName());
+
+		ObjectRelationship objectRelationship =
+			ObjectRelationshipLocalServiceUtil.addObjectRelationship(
+				null, _user.getUserId(),
+				_userSystemObjectDefinition.getObjectDefinitionId(),
+				_objectDefinition.getObjectDefinitionId(), 0,
+				ObjectRelationshipConstants.DELETION_TYPE_PREVENT,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				"relation1", false,
+				ObjectRelationshipConstants.TYPE_ONE_TO_MANY, null);
+
+		JaxRsApplicationDescriptor jaxRsApplicationDescriptor =
+			_userSystemObjectDefinitionManager.getJaxRsApplicationDescriptor();
+
+		String endpoint = StringBundler.concat(
+			jaxRsApplicationDescriptor.getApplicationPath(), StringPool.SLASH,
+			jaxRsApplicationDescriptor.getVersion(), "/openapi.json");
+
+		JSONAssert.assertEquals(
```

Comment:
I think it is not a good idea to include a strict validation here, as if other team modify the UserAccount schema or includes a new endpoint, our test will fail.

I think the best choice is to get rid of all the fields that do not depend on us, and just validating the fields that applies to the System Object case. That is, for example:

- Object1 schema with all the properties
- PageObject1 schema with all the properties
- UserAccount schema with the relationships
- All the relationship endpoints (get, put and delete)

And then, you can modify the type of comparison from STRICT to LENIENT (as the order does not matter either in the JSON file)

![image](https://github.com/user-attachments/assets/96731428-3cee-4f07-b1d6-dd8df169ac8b)

-

Reply:
I also miss the cases for the other relationships.

For example:

One UserAccount Many Object1 (`relation1ToM`)
Many UserAccount Many Object1 (`relationMtoM`)

Then, we could rename from `relation1` to `relationMto1` to match the pattern and provide more info of which property belongs to which relationship type